docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "2. Go to your **Account** (locate your profile avatar in the top-right corner of the page)\n",
+    "\n",
+    "![kaggle-account](images/kaggle_account.png)\n",
+    "\n",
+    "3. Scroll down your account page until you find the **API cell**.\n",
+    "4. Click on to \"Create new API token\" button.\n",
+    "\n",
+    "![kaggle-api-key](images/api_key.png)\n",
+    "\n",
+    "This will download a file called `kaggle.json`. This file contains the credentials required to use the Kaggle APIs. \n",
+    "\n",
+    "5. Open this file, and include this data into the form below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Setup `kaggle` API credentials { display-mode: \"form\" }\n",
+    "from ipywidgets import Password\n",
+    "from ipywidgets import widgets\n",
+    "from IPython.display import display\n",
+    "from IPython.display import Image\n",
+    "import os\n",
+    "import stat\n",
+    "import json\n",
+    "\n",
+    "kaggle_api_credentials = os.path.expanduser('~/.kaggle/kaggle.json')\n",
+    "\n",
+    "API_CREDENTIALS_EXIST = os.path.exists(kaggle_api_credentials) \n",
+    "\n",
+    "if not API_CREDENTIALS_EXIST:\n",
+    "    user = widgets.Text(\n",
+    "        placeholder='Insert your Kaggle User Name!',\n",
+    "        description='User Name',\n",
+    "        disabled=False\n",
+    "    )\n",
+    "\n",
+    "    pwd = Password(\n",
+    "        description='Key',\n",
+    "        placeholder='Insert key for user {}'.format(user.value)\n",
+    "    )\n",
+    "\n",
+    "    def store_kaggle_credentials(wdgt):\n",
+    "        folder = os.path.expanduser('~/.kaggle')\n",
+    "        os.makedirs(folder, exist_ok=True)\n",
+    "        file = '{}/kaggle.json'.format(folder)\n",
+    "        with open(file, 'w') as kaggle_file:\n",
+    "            json.dump({\n",
+    "                'username': user.value,\n",
+    "                'key': wdgt.value\n",
+    "            }, kaggle_file)\n",
+    "\n",
+    "            print(\"Credentials created in {} ✅\".format(folder))\n",
+    "        os.chmod(file, stat.S_IREAD | stat.S_IWRITE)\n",
+    "\n",
+    "    user.on_submit(lambda wdg: display(pwd))\n",
+    "    pwd.on_submit(store_kaggle_credentials)\n",
+    "\n",
+    "    display(user)\n",
+    "else:\n",
+    "    print(\"Kaggle API credentials found ✅ - All set, Ready to start!\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Authenticate on Kaggle via APIs { display-mode: \"form\" }\n",
+    "import kaggle\n",
+    "\n",
+    "kaggle.api.authenticate()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title **Search** on Kaggle for available Netflix-Prize datasets { display-mode: \"form\" }\n",
```

Comment:
Maybe put this cell and the following explanation in hidden section, with name something like "Searching dataset on Kaggle (optional)"

Reply:
done